cloud-energy-collection/iriscasttools/test/test_main.py

```diff
@@ -0,0 +1,56 @@
+"""
+Tests for main functions for iriscasttools package
+"""
+
+from unittest.mock import patch
+import pytest
+
+from main import get_iriscast_stats
+
+
+@pytest.mark.parametrize(
+    "test_csv_flag, test_include_header",
+    [(False, False), (True, True), (True, False)],
+)
+@patch("utils.get_ipmi_power_stats")
+@patch("utils.get_os_load")
+@patch("utils.get_ram_usage")
+@patch("utils.to_csv")
+def test_get_iriscast_stats(
+    mock_to_csv,
+    mock_get_ram_usage,
+    mock_get_os_load,
+    mock_get_ipmi_power_stats,
+    test_csv_flag,
+    test_include_header,
+):
+    """
+    Test get iriscast stats function
+
+    Keyword arguments:
+        - mock_get_ipmi_power_stats -- Mock obj for get_ipmi_power_stats
+        - test_poll_period -- int, intervals between each reading, used to calculate watt hours
+    """
+
+    expected_power_stats = {"current_power": ""}
+    expected_os_load_stats = {"os_load_5": ""}
+    expected_ram_stats = {"ram_usage_percentage": ""}
+
+    expected_all = dict(expected_power_stats)
+    expected_all.update(expected_os_load_stats)
+    expected_all.update(expected_ram_stats)
+
+    mock_get_ipmi_power_stats.side_effect = [expected_power_stats]
+    mock_get_os_load.return_value = expected_os_load_stats
+    mock_get_ram_usage.return_value = expected_ram_stats
+
+    _ = get_iriscast_stats(test_csv_flag, test_include_header)
```

Comment:
We should check that what `get_iriscast_stats` returns was the same as what one of the mock methods returned

Currently you could change it to `return None` and the test would still pass